### PR TITLE
Fix merge for deleted sorted index

### PR DESCRIPTION
**Pull request by sam-herman**
### Description
New off heap merge that was introduced in #87 was missing an edge case for graceful handling of deleted documents and out of order merges.
- Created a test that properly reproduced the scenario
- Issue was detected in OSB as well

### Check List
- [x ] New functionality includes testing.
- [x ] New functionality has been documented.
~- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).~
- [ x] Commits are signed per the DCO using `--signoff`.
~- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/knn/index/codec/jvector/JVectorWriter.java [hunk 1]
```diff
@@ -597,15 +619,32 @@ public RandomAccessMergedFloatVectorValues(
...
+                    if (docMaps[readerIdx].get(docId) == -1) {
+                        log.warn(
+                            "Document {} in reader {} is not mapped to a global ordinal from the merge docMaps. Will skip this document for now",
+                            docId,
+                            readerIdx
+                        );
+                    } else {
+                        // Mapping from global ordinal to [readerIndex, readerOrd]
+                        final int globalOrd = docMaps[readerIdx].get(docId);
+                        ordMapping[globalOrd][0] = readerIdx; // Reader index
```
[the remark this answers is not in the document]
Comment (sam-herman) [reply src/main/java/org/opensearch/knn/index/codec/jvector/JVectorWriter.java:631]: actually never mind, will fix it now with parameter (`READER_ID`), better to do now in case follow up is delayed.
